lib/repository/movie_repository.dart

```diff
@@ -0,0 +1,10 @@
+import 'movie_data_class.dart';
+import 'movie_genres_data_class.dart';
+
+abstract class MovieRepository {
+  Future<List<MovieFile>> getMovies();
+
+  Future<List<MovieGenres>> getGenres();
+
+  Future<List<String>> getGenresByIds(List<int> ids);
```

Comment:
Why do you need to define this method as a Future?

Reply:
I need Future because getGenresByIds() calls getGenres() which needs future due to rootBundle.loadString() returns a Future<String>